src/kitovu/utils.py

```
@@ -22,51 +23,49 @@ def get_password(plugin: str, identifier: str) -> str:
     return password
 
 
-class Error(Exception):
+class SchemaValidator:
+    """A validator for creating and merging errors in schema definitions."""
+
+    def __init__(self, abort: bool = True) -> None:
+        self.errors: typing.List[str] = []
+        self.abort: bool = abort
```

I don't understand why we need two ways of signalling errors - can't we always do what we do with `abort=True` and let the caller handle the exception? You already get multiple erors in the `InvalidSettingsErrors` exception (and the message and all), so why duplicate all that functionality again with the `valid` property and `abort=False`?

---

The global Settings file should raise an exception, but the the plugins shouldn't (otherwise you'd only get the errors of the first invalid plugin).

I've started with handling it in the main method, but it just gets confusing and both the `config_error` and `start` methods would need to implement it. This way the `SchemaValidator` class is responsible for merging and preparing the error messages